aws/resource_aws_iam_saml_provider.go

```diff
@@ -108,22 +107,23 @@ func resourceAwsIamSamlProviderUpdate(d *schema.ResourceData, meta interface{})
 }
 
 func resourceAwsIamSamlProviderDelete(d *schema.ResourceData, meta interface{}) error {
-	iamconn := meta.(*AWSClient).iamconn
+	conn := meta.(*AWSClient).iamconn
 
 	input := &iam.DeleteSAMLProviderInput{
 		SAMLProviderArn: aws.String(d.Id()),
 	}
-	_, err := iamconn.DeleteSAMLProvider(input)
+	_, err := conn.DeleteSAMLProvider(input)
 
 	return err
 }
 
-func extractNameFromIAMSamlProviderArn(arn, partition string) (string, error) {
-	// arn:aws:iam::123456789012:saml-provider/tf-salesforce-test
-	r := regexp.MustCompile(fmt.Sprintf("^arn:%s:iam::[0-9]{12}:saml-provider/(.+)$", partition))
-	submatches := r.FindStringSubmatch(arn)
-	if len(submatches) != 2 {
-		return "", fmt.Errorf("Unable to extract name from a given ARN: %q", arn)
+func extractNameFromIAMSamlProviderArn(samlArn string) (string, error) {
+	parsedArn, err := arn.Parse(samlArn)
+	if err != nil {
+		return "", fmt.Errorf("Unable to extract name from a given ARN: %q", parsedArn)
```

Comment:
Nit: in the event of error, i think passing the `samlArn` here in the error message will be helpful as `parsedArn` will be empty

Reply:
Missed that, will fix

Reply:
fixed